sdks/java/io/cassandra/src/main/java/org/apache/beam/sdk/io/cassandra/CassandraIO.java

```diff
@@ -1116,6 +1163,18 @@ private static Cluster getCluster(
           new QueryOptions().setConsistencyLevel(ConsistencyLevel.valueOf(consistencyLevel.get())));
     }
 
+    SocketOptions socketOptions = new SocketOptions();
+
+    builder.withSocketOptions(socketOptions);
+
+    if (connectTimeout != null) {
```

Comment:
no more null check needed if both timeouts are set as part of Read and Write and if you add the validation of inputs in the with* methods

Reply:
I was wrong: null tests are needed because there is no default value. If the user to not specify timeouts, then the valueProvider will be null.